mood-analysis.rb

```diff
@@ -28,5 +29,60 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+def strip_punctuation(words)
+  # 2. To make the results a little more accurate, let's write and utilize a method called `strip_punctuation` to 
+  # strip out the punctuation that affects the results. Namely, remove  exclamation marks (!), periods (.), commas (,), 
+  # and hashtags (#).
+  words.gsub(/[^\/A-Za-z0-9\s]/i, "")
+end
+
+# "smiley collector": iterate a corpus, analyze its mood, store return in an array.
+def print_all_moods(words)
+  words.collect { |statement| statement[0..4] + " " + analyze_mood(statement)}
+end
+
+def happy_days(words)
+  # Write a method called `happy_days` to determine how many logged entries it takes until there have 
+  # been three :-) happy days.
+
+  total_days = 0
+  hap_days = 0
+
+  words.each do |statement|
+    total_days += 1
+    if analyze_mood(strip_punctuation(statement)) == ":-)"
+      hap_days += 1
+    end
+    break if hap_days == 3
+  end
+
+  return "It takes #{total_days} days for 3 happy days to occur." if hap_days == 3
+  return "There were not 3 happy days."
+
+end
+
+def overall_mood(words)
+  feels = {
+          happy: [],
+          sad: [],
+          meh: []
+  }
+
+  words.each do |statement|
+    if analyze_mood(statement) == ":-)"
+      feels[:happy] << ":-)"
+    elsif analyze_mood(statement) == ":-("
+      feels[:sad] << ":-("
+    else
+      feels[:meh] << ":-|"
+    end
+  end
+
+  feels = feels.max_by { |key, value| feels.values.length}
+  "The over all mood is #{feels.first.to_s}."
+
+end
```

Comment:
Same here, I would have liked to see this method only return the emoticon itself and then outside the method to use that information in a puts.